manifests/00-crd-extension-console-log-link.yaml

```diff
@@ -0,0 +1,54 @@
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  name: consoleloglinks.console.openshift.io
+  annotations:
+    displayName: ConsoleLogLinks
+    description: Extension for customizing OpenShift web console log links
+spec:
+  scope: Cluster
+  group: console.openshift.io
+  versions:
+  - name: v1
+    served: true
+    storage: true
+  names:
+    plural: consoleloglinks
+    singular: consoleloglink
+    kind: ConsoleLogLink
+    listKind: ConsoleLogLinkList
+  additionalPrinterColumns:
+  - name: Text
+    type: string
+    JSONPath: .spec.text
+  - name: HrefTemplate
+    type: string
+    JSONPath: .spec.hrefTemplate
+  - name: Age
+    type: date
+    JSONPath: .metadata.creationTimestamp
+  subresources:
+    status: {}
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          type: object
+          description: Represents console log link customizations spec
+          required:
+          - text
+          - hrefTemplate
+          properties:
+            text:
+              type: string
+              description: Text of the link
+            hrefTemplate:
+              type: string
+              description: >-
+                Absolute secure URL (must use https) for the log link including variables to be replaced.
+                Supported variables include ${resourceName}, ${resourceUID}, ${containerName}, ${resourceNamespace}, and ${podLabels} and must match exactly.
+                e.g., https://example.com/logs?resourceName=\${resourceName}&containerName=\${containerName}&resourceNamespace=\${resourceNamespace}&podLabels=\${podLabels}
+              pattern: '^https://'
+            namespaceFilter:
```

Comment:
Trying to decide if we want a pattern on this. For example:
`^openshift-` vs `/^openshift-/` vs `/^openshift-/g` 

We are using `string.search(regex)`, so `^openshift-` will break, but the other two will work. [This stack overflow](https://stackoverflow.com/questions/172303/is-there-a-regular-expression-to-detect-a-valid-regular-expression) says that this is a perfect regex to check for a regex:

`/^((?:(?:[^?+*{}()[\]\\|]+|\\.|\[(?:\^?\\.|\^[^\\]|[^\\^])(?:[^\]\\]+|\\.)*\]|\((?:\?[:=!]|\?<[=!]|\?>)?(?1)??\)|\(\?(?:R|[+-]?\d+)\))(?:(?:[?+*]|\{\d+(?:,\d*)?\})[?+]?)?|\|)*)$/
`
I kid about including that 😄 

To make cross reference easy: https://github.com/openshift/console/pull/1892/files#r304608136

We may want to do something, however? At least check for leading `/`?  I don't think we should necessarily try to be comprehensive, that sounds like bugs.

Reply:
Per discussion in https://github.com/openshift/api/pull/385, leading `/` is not necessary. I'm not sure what validation is really practical here. I think this is reasonable for now, and we can improve validation as a bug fix if needed.

Reply:
Works for me.